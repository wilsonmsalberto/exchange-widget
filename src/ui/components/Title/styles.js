import { css } from '@emotion/core';
import styled from '@emotion/styled';

export const Heading = styled.div`
  ${(props) => {
    switch (props.as) {
      case 'h1':
      default:
        return css`
          font-size: 4rem;
        `;
      case 'h2':
        return css`
          font-size: 3rem;
        `;
    }
  }}

  padding: 1rem;
  text-align: center;
  font-weight: 300;
`;

export default Heading;
