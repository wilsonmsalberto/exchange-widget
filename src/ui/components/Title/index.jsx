import React from 'react';
import PropTypes from 'prop-types';

import { Heading } from './styles';

const Title = ({ children, as }) => <Heading as={ as }>{children}</Heading>;

Title.propTypes = {
  as: PropTypes.string,
  children: PropTypes.node.isRequired,
};

Title.defaultProps = {
  as: undefined,
};

export default Title;
