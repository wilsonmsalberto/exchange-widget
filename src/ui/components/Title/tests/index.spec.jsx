import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { matchers } from 'jest-emotion';

import Title from '..';

expect.extend(matchers);

const getRenderWrapper = (props) => render(<Title { ...props } />);

describe('Title Component', () => {
  describe('Rendering', () => {
    it('should render properly', () => {
      const wrapper = getRenderWrapper({ children: 'Example Heading' });

      const { asFragment } = wrapper;

      expect(asFragment()).toMatchSnapshot();
    });

    it('should contain a node content', () => {
      const MockedNode = <div>Node Content</div>;
      const wrapper = getRenderWrapper({ children: MockedNode });

      const { asFragment } = wrapper;

      expect(asFragment()).toMatchSnapshot();
    });

    it('should render a <h1> title', () => {
      const wrapper = getRenderWrapper({
        children: 'H1 Title',
        as: 'h1',
      });

      const { getByText } = wrapper;

      const title = getByText('H1 Title');
      const titleElement = wrapper.container.getElementsByTagName('h1')[0];

      expect(title).toHaveStyleRule('font-size', '4rem');
      expect(titleElement).toBeDefined();
    });

    it('should render a <h2> title', () => {
      const wrapper = getRenderWrapper({
        children: 'H2 Title',
        as: 'h2',
      });

      const { getByText } = wrapper;

      const title = getByText('H2 Title');
      const titleElement = wrapper.container.getElementsByTagName('h2')[0];

      expect(title).toHaveStyleRule('font-size', '3rem');
      expect(titleElement).toBeDefined();
    });
  });
});
