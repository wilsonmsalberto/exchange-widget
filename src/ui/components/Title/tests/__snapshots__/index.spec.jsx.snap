// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Component Rendering should contain a node content 1`] = `
<DocumentFragment>
  <div
    class="css-nr18n0"
  >
    <div>
      Node Content
    </div>
  </div>
</DocumentFragment>
`;

exports[`Title Component Rendering should render properly 1`] = `
<DocumentFragment>
  <div
    class="css-nr18n0"
  >
    Example Heading
  </div>
</DocumentFragment>
`;
